File: Кросс/Lab3/src/Task1.java
import java.util.Scanner;
import java.math.BigInteger;

public class Task1 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Введіть значення x:");
        double x = scanner.nextDouble();
        scanner.close();

        double s = 0;
        double eps = 0.00001;
        int k = 0;
        double u = Math.pow(x, k) / factorial(k).doubleValue();
        while (Math.abs(u) > eps) {
            BigInteger factorialResult = factorial(k);
            s += u;
            k++;
            u = Math.pow(x, k) / factorial(k).doubleValue();
        }

        System.out.printf("Значення суми s = %.6f\n", s);
        System.out.printf("Значення функції: %.6f\n", Math.pow(Math.E, x));
        System.out.println("Кількість доданків " + (k));
    }

    public static BigInteger factorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("Factorial is not defined for negative numbers");
        }

        BigInteger result = BigInteger.ONE;
        for (int i = 1; i <= n; i++) {
            result = result.multiply(BigInteger.valueOf(i));
        }

        return result;
    }
}

File: Кросс/Lab3/src/Task1.kt
import java.math.BigInteger
import java.util.*
import kotlin.math.pow

fun main() {
    val scanner = Scanner(System.`in`)
    println("Введіть значення x:")
    val x = scanner.nextDouble()
    scanner.close()

    var s = 0.0
    val eps = 0.00001
    var k = 0
    var u = Math.pow(x, k.toDouble()) / factorial(k).toDouble()

    while (Math.abs(u) > eps) {
        val factorialResult = factorial(k)
        s += u
        k++
        u = Math.pow(x, k.toDouble()) / factorial(k).toDouble()
    }

    println("Значення суми s = %.6f".format(s))
    println("Значення функції: %.6f".format(Math.E.pow(x)))
    println("Кількість доданків $k")
}

fun factorial(n: Int): BigInteger {
    require(n >= 0) { "Factorial is not defined for negative numbers" }

    var result = BigInteger.ONE
    for (i in 1..n) {
        result = result.multiply(BigInteger.valueOf(i.toLong()))
    }

    return result
}
